src/RustyObjectStore.jl

```diff
@@ -588,6 +589,9 @@ function rust_message_to_reason(msg::AbstractString)
         return UnknownError()
     end
 end
+struct DeleteException <: RequestException
```

Comment:
I think it would be good to also have the reason on this exception, just like GetException, in order for it to be able to implement the `reason` method, because currently we assume all `<: RequestException` implement it